### PR TITLE
This branch added smart contracts for temperature violations

### PR DESCRIPTION
**Pull request by hulk-baba**

### CODE REVIEW

File: chaincode/fabcar/fabcar.go [hunk 1]
```diff
@@ -130,17 +138,22 @@ func (s *SmartContract) createMed(APIstub shim.ChaincodeStubInterface, args []st
...
-		BatchNo: objArgs[2],
-		ExpiryDate: objArgs[3],
-		Owner: objArgs[4],
-		Location: objArgs[5],
-		Timestamp: time.Now(),
-		Temperature: objArgs[6],
-	}
-	return med
+	if n, err := strconv.ParseFloat(objArgs[6], 64); err == nil {
+        med := Med{
```
Comment (sriyanshcse) [inline chaincode/fabcar/fabcar.go:181]: Why initialised whole object again. Changing just the temperature field should be just fine.

File: chaincode/fabcar/fabcar.go [hunk 2]
```diff
@@ -204,22 +217,98 @@ func (s *SmartContract) changeMedOwner(APIstub shim.ChaincodeStubInterface, args
...
+		APIstub.PutState(args[0], medAsBytes)
+		return shim.Success(nil)
+    }
 	return shim.Success(nil)
 }
 
+
+
+
+func (s *SmartContract) tempartureViolations(APIstub shim.ChaincodeStubInterface, args []string) sc.Response {
```
Comment (sriyanshcse) [inline chaincode/fabcar/fabcar.go:294]: Is this Still Needed ?

Comment (hulk-baba) [reply chaincode/fabcar/fabcar.go:294]: This function does not hurt

File: chaincode/fabcar/fabcar.go (hunk 1 above)
Comment (hulk-baba) [reply chaincode/fabcar/fabcar.go:181]: it is not initialised again, it is created only when we have successfully parsed temperature argument as the float.

**Pull request merged**